database/dataGeneration.js

```diff
@@ -0,0 +1,133 @@
+
+const tags = ['Good for groups', 'Desserts', 'Appetizers', 'Drinks', 'Kid friendly'];
+
+const circleColors = ['#6c8ae4', '#d86441', '#bb6acd', '#df4e96'];
+
+const jabber = new Jabber(buzzWords, 2);
+
+const allWritesFinished = [false, false];
+
+
+
+async function createReviewersAndReviews(NumberOfPeople) {
+  let reviewerCount = NumberOfPeople || 100;
+  const writeReviews = fs.createWriteStream('Reviews.csv');
+  const writeReviewers = fs.createWriteStream('Reviewers.csv');
+  let ok = true;
+  let reviewIndex = 0;
+
+
+  writeReviewers.write('id, color, vip, reviewCount, firstName, lastName \n');
+  writeReviews.write(`id, restaurantId, reviewerId overall, food, service, ambience, dineDate, noise, recommend, comments, filterTag \n`)
+
+  writeReviewerToCSV();
+  async function writeReviewerToCSV() {
+    do {
+      let reviewer = {};
+      reviewer.id = reviewerCount;
+      reviewer.reviewCount = faker.random.number({min: 1, max: 15})
+      reviewer.color = faker.random.arrayElement(circleColors);
+      reviewer.vip = Math.random() < 0.3;
+      reviewer.firstName = (Math.random() < 0.7) ? faker.name.firstName() : '';
+      if (reviewer.firstName.length) {
+        reviewer.lastName = (Math.random() < 0.7) ? faker.name.lastName() : '';
+      } else {
+        reviewer.lastName = '';
+      }
+      reviewerCount -= 1; // maybe move to below the review section?
+
+      ///////////////////////////////////////////
+
+      reviewIndex += await writeReview(reviewer.id, reviewer.reviewCount, writeReviews, reviewIndex);
+
+      if (reviewerCount === -1) {
+        writeReviewers.end();
+      } else {
+        ok = writeReviewers.write(
+          `${reviewer.id}, ${reviewer.color}, ${reviewer.vip}, ${reviewer.reviewCount}, ${reviewer.firstName}, ${reviewer.lastName} \n`
+        );
+      }
+    } while (reviewerCount >= 0 && ok);
+
+    if (reviewerCount > 0) {
+      writeReviewers.once('drain', writeReviewerToCSV);
+    }
+  }
+};
+
+async function writeReview(reviewerId, numberOfReviews, writeReviews, reviewIndex){
+
+  let reviewOk = true;
+  writeReviewToCSV();
+
+  function writeReviewToCSV() {
+
+    do {
+      let review ={};
+      review.restaurantId = Math.floor(Math.random() * 10000000);
+      review.reviewerId = reviewerId;
+      review.id = reviewIndex;
+      review.overall = faker.random.number({min:1, max: 5});
+      review.food = faker.random.number({min:1, max: 5});
+      review.service = faker.random.number({min:1, max: 5});
+      review.ambience = faker.random.number({min:1, max: 5});
+      review.dineDate = moment(faker.date.between('2019-01-01', '2020-01-24')).format('YYYY-MM-DD');
+      review.noise = faker.random.number({min:1, max: 5});
+      review.recommend = Math.random() < 0.7;
+      review.comments = jabber.createParagraph(faker.random.number({min:10, max: 50}));
+      review.filterTag = faker.random.arrayElement(tags);
+      reviewIndex += 1;
+
```

Comment:
A personal preference, but this is hard to read. Maybe taking some of your values and setting them to variables would help, but overall the logic looks great.

Reply:
Alright, I'll look into it!